live-build/misc/migration-scripts/dx_delete

```diff
@@ -39,10 +39,94 @@ function die() {
 # names are randomly generated.
 #
 
-linux_dataset="rpool/ROOT"
-if ! zfs list ${linux_dataset} &>/dev/null; then
-	echo "Linux dataset '${linux_dataset}' is not installed."
+LX_DATASET="rpool/ROOT"
+if ! zfs list ${LX_DATASET} &>/dev/null; then
+	echo "Linux dataset '${LX_DATASET}' is not installed."
 	exit 0
 fi
 
-zfs destroy -r ${linux_dataset} || die "Failed to destroy Linux dataset '${linux_dataset}'"
+LX_RDS_PARENT=$(zfs list -o name -H -d 1 "$LX_DATASET" | tail -n 1)
+[[ -n $LX_RDS_PARENT ]] || die "could not find Linux RDS parent dataset"
+LX_CONTAINER="${LX_RDS_PARENT##*/}"
+LX_RDS_TMP_ROOT_MOUNT="/tmp/$LX_CONTAINER/root"
+
+MDS_SNAPNAME="MDS-CLONE-upgradeverify"
+MDS_CLONE=domain0/$MDS_SNAPNAME
+SVC=svc:/system/delphix/postgres:$MDS_SNAPNAME
+PG_DATA=/$MDS_CLONE/db
+
+function dx_execute_cleanup() {
+	#
+	# Undo any existing default timeout commands in the bootloader's menu.
+	#
+	sed -i '/menu_timeout_command/d' /boot/menu.rc.local
+	[[ "$(grep -cF 'menu_timeout_command' /boot/menu.rc.local)" -eq 0 ]] ||
+		die "failed to cleanup previous default command"
+
+	#
+	# We could also delete /var/delphix/migration here but we decide
+	# to leave it for debugging purposes. Re-executing dx_execute always
+	# deletes and recreates the directory so we'll always have the logs
+	# of the latest run.
+	#
+}
+
+function dx_apply_cleanup() {
+	#
+	# Remove new bootloader entry
+	#
+	sed -i '/set lxcmdline/d' /boot/menu.rc.local
+	[[ "$(grep -cF 'set lxcmdline' /boot/menu.rc.local)" -eq 0 ]] ||
+		die "failed to cleanup lxcmdline from previous run"
+}
+
+function dx_verify_cleanup() {
+	#
+	# dx_verify creates a clone of the MDS dataset and the postgres
+	# service as well as a clone of /var/delphix for masking validation.
+	# If any are left, clean them up. (This logic is based on dx_manage_pg
+	# stop and cleanup.)
+	#
+	local sta
+	sta=$(svcs -Ho sta $SVC)
+	if [[ -n "$sta" && ("$sta" == ON || "$sta" == DGD) ]]; then
```

Comment:
Yeah, I don't see why dx_manage_pg was only deleting the service in those states - it makes to me to delete fully. I found from experimenting that even if the service is `OFF`, it still causes the next round of verification to fail since you can't create a new one.